src/packagedcode/cargo.py

```diff
@@ -70,33 +71,46 @@ def get_package_root(cls, manifest_resource, codebase):
         return manifest_resource.parent(codebase)
 
     def repository_homepage_url(self, baseurl=default_web_baseurl):
+        if not self.name:
+            return None
         return '{}/crates/{}'.format(baseurl, self.name)
 
     def repository_download_url(self, baseurl=default_download_baseurl):
+        if not self.name and not self.version:
+            return None
         return '{}/crates/{}/{}/download'.format(baseurl, self.name, self.version)
 
     def api_data_url(self, baseurl=default_api_baseurl):
+        if not self.name:
+            return None
         return '{}/crates/{}'.format(baseurl, self.name)
 
 
 def is_cargo_toml(location):
     return (filetype.is_file(location) and fileutils.file_name(location).lower() == 'cargo.toml')
 
 
+def is_cargo_lock(location):
+    return (filetype.is_file(location) and fileutils.file_name(location).lower() == 'cargo.lock')
+
+
 def parse(location):
     """
-    Return a Package object from a Cargo.toml file or None.
+    Return a Package object from a Cargo.toml/Cargo.lock file or None.
     """
-    if not is_cargo_toml(location):
+    if not is_cargo_toml(location) and not is_cargo_lock(location):
```

Comment:
What about this instead?
```
handlers = {'cargo.toml': build_cargo_toml_package, 'cargo.lock': build_cargo_lock_package}
filename = filetype.is_file(location) and fileutils.file_name(location).lower()
handler = handlers.get(filename)
if handler:
    return handler(toml.load(location, _dict=OrderedDict))
```

Reply:
This also looks good. I add this code.

Reply:
You could even use this pythonic construct:
```
return handler and handler(toml.load(location, _dict=OrderedDict))
```

Reply:
Done